pkg/apis/workflow/v1alpha1/common_test.go

```diff
@@ -1,22 +1,37 @@
 package v1alpha1
 
 import (
+	"sort"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 )
 
 func TestCompareByPriority(t *testing.T) {
 
-	stepOne := WorkflowStep{Name: "A", Priority: 100}
-	stepTwo := WorkflowStep{Name: "B", Priority: 50}
+	t.Run("SortingByPriority", func(t *testing.T) {
+		stepGroup := []WorkflowStep{{Name: "A", Priority: 10}, {Name: "B", Priority: 1000}, {Name: "C", Priority: 100}, {Name: "D", Priority: 10000}}
 
-	// true since priority of A is greater than B
-	assert.True(t, CompareByPriority(&stepOne, &stepTwo))
+		sort.Slice(stepGroup, func(i, j int) bool {
+			return CompareByPriority(&stepGroup[i], &stepGroup[j])
+		})
 
-	stepOne.Priority = 0
-	stepTwo.Priority = 0
+		assert.Equal(t, "D", stepGroup[0].Name)
+		assert.Equal(t, "B", stepGroup[1].Name)
+		assert.Equal(t, "C", stepGroup[2].Name)
+		assert.Equal(t, "A", stepGroup[3].Name)
+	})
 
-	assert.False(t, CompareByPriority(&stepOne, &stepTwo))
+	t.Run("SortingByName", func(t *testing.T) {
+		stepGroup := []WorkflowStep{{Name: "Banana"}, {Name: "Durian"}, {Name: "Apple"}, {Name: "Cherry"}}
 
+		sort.Slice(stepGroup, func(i, j int) bool {
+			return CompareByPriority(&stepGroup[i], &stepGroup[j])
+		})
+
+		assert.Equal(t, "Durian", stepGroup[0].Name)
+		assert.Equal(t, "Cherry", stepGroup[1].Name)
+		assert.Equal(t, "Banana", stepGroup[2].Name)
+		assert.Equal(t, "Apple", stepGroup[3].Name)
```

Comment:
in the original version, the sort is in alphabetical order, I think we shouldn't change that behavior